lib/buffer.c

```diff
@@ -199,6 +199,50 @@ buffer_putstr (struct buffer *b, const char *c)
   buffer_put(b, c, strlen(c));
 }
 
+/* Expand \n to \r\n */
+void
+buffer_put_crlf(struct buffer *b, const void *origp, size_t origsize)
+{
+  struct buffer_data *data = b->tail;
+  const char *p = origp, *end = p + origsize, *lf;
+  size_t size;
+
+  lf = memchr(p, '\n', end - p);
```

Comment:
Should there be a null return check here to handle the case where there is no '\n' in the string?